spec/requests/categories/create_spec.rb

```diff
@@ -0,0 +1,36 @@
+RSpec.describe 'Category create', type: :request do
+  let(:category) { create(:category) }
+
+  context 'with valid params' do
+    let(:params) do
+      { category: { name: :name } }
+    end
+
+    it 'creates category ' do
+      expect do
+        post '/categories', params: params
+      end.to change(Category, :count).by(1)
+
+      expect(response).to redirect_to(categories_path)
+      follow_redirect!
+
+      category = Category.last.name
+      expect(category).to be_a(String)
+      expect(response.body).to include(category)
+      expect(response.body).to include('Категория &quot;name&quot; добавлена')
+    end
+  end
+
+  context 'with invalid params' do
+    let(:params) do
+      { category: { name: '' } }
+    end
+
+    it 'does not create category' do
+      expect do
+        post '/categories', params: params
+      end.not_to change(Category, :count)
+      expect(response.body).to include('Категория &quot;&quot; не может быть добавлена')
```

Comment:
I guess you need to check that correct validation error present on form. This test is fully incorrect because you're rendering `index` template for invalid category.

```suggestion
      end.not_to change(Category, :count)
      
      expect(response.body).to include('Категория &quot;&quot; не может быть добавлена')
```